facia-purger/src/main/scala/com/gu/purge/facia/Lambda.scala

```diff
@@ -24,22 +21,17 @@ class Lambda extends RequestHandler[S3Event, String] {
 
     println(s"Processing ${entities.size} updated entities ...")
 
-    entities.foreach { entity =>
-      new ParseS3Path(STAGE, entity.getObject.getKey).apply() map { id =>
-        sendPurgeRequest(id)
-      }
+    entities.forall { entity =>
+      new ParseS3Path(stage, entity.getObject.getKey).apply().map(sendPurgeRequest).isDefined
```

Comment:
nitpick: Usually `map` implies no side effect. In case of a function with side effect, it is the norm to use `foreach`